### PR TITLE
feat(validator): Add workflow for validator tests

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR introduces a workflow to execute validator tests on CI Additionally, it includes a Makefile within `e2e/tools/validator`, enabling the execution of targets such as `make fmt` and `make test` for local testing

### CODE REVIEW

**Comment by vprashar2929** [comment]
Sample CI run: https://github.com/sustainable-computing-io/kepler/actions/runs/9683848586/job/26720262220?pr=1570

File: Makefile [hunk 1]
```diff
@@ -286,6 +286,13 @@ unit-test: ginkgo-set tidy-vendor ## Run unit tests.
 		--race --count=1 \
 		$(TEST_PKGS)
 
+.PHONY: validator-test
+validator-test: ## Run validator tests.
+	@echo "Running validator tests..."
+	@cd ./e2e/tools/validator && \
+	hatch env create && \
+	hatch run test || { echo 'validator-test failed'; exit 1; }
```
Comment (sthaha) [inline Makefile:294]:
```suggestion
	hatch run test
```

File: .github/workflows/validator_test.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      # checkout source code
+      - name: checkout source
+        uses: actions/checkout@v4.1.1
+      # setup Python environment
+      - name: Set up Python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: "3.12"
```
Comment (sthaha) [inline .github/workflows/validator_test.yml:17]:
```suggestion
          python-version: "3.11"
```

File: Makefile [hunk 3]
```diff
@@ -286,6 +286,13 @@ unit-test: ginkgo-set tidy-vendor ## Run unit tests.
 		--race --count=1 \
 		$(TEST_PKGS)
 
+.PHONY: validator-test
+validator-test: ## Run validator tests.
+	@echo "Running validator tests..."
+	@cd ./e2e/tools/validator && \
+	hatch env create && \
```
Comment (sthaha) [inline Makefile:293]: I think it is better to have a `validator-env` target that `test` depends on

File: .github/workflows/validator_test.yml (hunk 2 above)
Comment (vprashar2929) [reply .github/workflows/validator_test.yml:17]: Do we have any hard requirements on `3.11` ?

Comment (sthaha) [reply .github/workflows/validator_test.yml:17]: Indeed we do :( .. rhel 9

Comment (sthaha) [reply .github/workflows/validator_test.yml:17]: I have pinned the python version in #1573

**Comment by KaiyiLiu1234** [comment]
This is great to add to the github actions workflow. The unit tests however need some more work last time I checked. Maybe we can improve it? @vprashar2929 I can raise it as an issue and hop on that if you would like.

File: Makefile (hunk 3 above)
Comment (sthaha) [reply Makefile:293]: @vprashar2929 , gentle reminder for a separate makefile for Validator :) but actually think `hatch` makes it really unnecessary to have a Makefile

**Review by sthaha — CHANGES_REQUESTED** [review]

**Review by sthaha — APPROVED** [review]

**Pull request merged**